app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    def wrapper(*args) -> Callable:
+        results_item = (func, args)
+        if results_item in results:
+            print("Getting from cache")
+            return results[results_item]
+        else:
+            print("Calculating new result")
+            new_result = func(*args)
+            results[results_item] = new_result
+            return new_result
```

Comment:
In the else branch, you calculate the new result, store it in the 'results' dictionary and return it. However, the 'print' statement is not necessary and should be removed. It's a part of the debugging process and should not be present in the final code.